internal/olm/client/status.go

```diff
@@ -71,13 +87,74 @@ func (c Client) GetObjectsStatus(ctx context.Context, objs ...runtime.Object) St
 	return Status{Resources: rss}
 }
 
-func (s Status) HasExistingResources() bool {
+// HasInstalledResources only returns true if at least one resource in s
+// was returned successfully by the API server. A resource error status
+// containing any error except "not found", or "no kind match" errors
+// for Custom Resources, will result in HasInstalledResources returning
+// false and the error.
+func (s Status) HasInstalledResources() (bool, error) {
+	crdKindSet, err := s.getCRDKindSet()
+	if err != nil {
+		return false, fmt.Errorf("error getting set of CRD kinds in resources: %w", err)
+	}
 	for _, r := range s.Resources {
 		if r.Resource != nil {
-			return true
+			return true, nil
+		} else if r.Error != nil && !apierrors.IsNotFound(r.Error) {
+			nkmerr := &meta.NoKindMatchError{}
+			if !errors.As(r.Error, &nkmerr) || !crdKindSet.Has(r.GVK.Kind) {
+				return false, r.Error
+			}
 		}
 	}
```

Comment:
It is but shouldn't be dependent on order. Sorting by `r.Error == nil` (installed resources sorted first) should suffice, since we want to report if any resources are installed.